(Feature): Allow an additional JS file to be passed to SC to have it …

IngwiePhoenix opened the pull request:
…modify the global scope.

This is essentially useful for:
- Using JavaScript dialects in workers.
- Applying various patches to the global scope.
- Short-circuiting the application if a cruicial error is found.
- Sharing initializing code with Broker processes, reducing duplicated code.

The proper ioCluster and nData PRs are on their way.

Also, within the samples, the require statement was changed to require the next top-level folder. I saw this iseful for testing, but I'd forgotten to change it back once I was done. Feel free to leave it in if you'd like.

jondubois commented:
@IngwiePhoenix I'm on holiday this week but I read through it and it looks good.
I will merge it once I've had time to test it. Feel free to make the PR for nData in the meantime - That way I could test everything together.

Nice work!

IngwiePhoenix commented:
@jondubois I did, on ioCluster, nData and SC. :) All you gotta do is test it.
And, thanks! I am depending on this change, it's why im pushing so much for it, to be honest. :)

jondubois commented:
@IngwiePhoenix Testing and merging this is top of my TODO list for when I get back from my holiday in a few days ;p

jondubois commented:
@IngwiePhoenix Great work. All merged. SC v2.3.22.

IngwiePhoenix commented:
Aaaawesome! :)

Going to update my project to use ^2.3.22 and I should be all good.

Glad it worked out =)

jondubois commented:
@IngwiePhoenix I ended up removing the init controller boilerplate logic from the sample app in v2.3.24 - This is because it's a more advanced feature and we don't want to overwhelm the user with too much stuff up front. We'll have to mention the initController in the docs on the website though.

Feel free to make PR for that too (I will do it if you don't have the time though). The page on the website is here https://github.com/SocketCluster/socketcluster-website/blob/master/public/app/views/docs/api-socketcluster.html

IngwiePhoenix commented:
Alright - works for me. Just decided to add it in, for testing purposes as well. That is how I tested the xhanges, actually. :)

IngwiePhoenix commented:
I’ll take care of writing the doc page. I have a lot of time on my hands, so I don’t mind!

SC is an awesome project, so contributing to it is really awesome. :) I don’t get to put something up to a nice project very often.

IngwiePhoenix commented:
And there you go!

I was surprised to find myself looking at AngularJS HTML. I haven'T used Angluar, ever. Anyway, I added some documentation about it. I moved the special note to the bottom so it doesn't jumpscare the user right away.

IngwiePhoenix commented:
@jondubois Still waiting for merging:
https://github.com/SocketCluster/socketcluster-website/pull/4

Just letting you know in case you'd overseen it. :)
